gh-106915: Add PyImport_ImportOrAddModule() function (#111559)
python/cpython: Doc/c-api/import.rst

   an exception set on failure (the module still exists in this case).

.. c:function:: int PyImport_ImportOrAddModule(const char *name, PyObject **module)

>    Get an already imported module from :data:`sys.modules`. If *name* is not
>    :data:`sys.modules`, create a new module and store it in
>    :data:`sys.modules`.

Reviewer: ```suggestion
   Get an already imported module, or create an empty new one.
```

>    - If the module was already imported, set *\*module* to a :term:`strong

Reviewer: ```suggestion
   - If the module name is already present in from :data:`sys.modules`, set *\*module* to a :term:`strong
```

Author: I replaced "in from sys.modules" with "in sys.modules".

     reference` to the existing module, and return 0.

>    - If the module does not exist, create a module, store it in

Reviewer: ```suggestion
   - If the module does not exist in :data:`sys.modules` yet, create a new empty module, store it in
```
